tools/sigclient/pkg/query/validation.go

```diff
@@ -597,36 +602,60 @@ func ValidateStatsQueryResults(queryRes *Result, expRes *Result) error {
 	sortMeasureResults(queryRes.MeasureResults)
 	sortMeasureResults(expRes.MeasureResults)
 
+	verifyGroupByValues := !expRes.DoNotVerifyGroupByValues
+
+	if !verifyGroupByValues {
+		log.Infof("ValidateStatsQueryResults: Skipping GroupByValues Elements match.")
+	}
+
 	for idx, expMeasureRes := range expRes.MeasureResults {
 		var err error
 		queryMeasureRes := queryRes.MeasureResults[idx]
 
-		equal := reflect.DeepEqual(queryMeasureRes.GroupByValues, expMeasureRes.GroupByValues)
-		if !equal {
-			return fmt.Errorf("ValidateStatsQueryResults: GroupByCombination mismatch, expected: %+v, got: %+v", expMeasureRes.GroupByValues, queryMeasureRes.GroupByValues)
+		if len(queryMeasureRes.GroupByValues) != len(expMeasureRes.GroupByValues) {
+			return fmt.Errorf("ValidateStatsQueryResults: GroupByValues length mismatch, expected: %v, got: %v", len(expMeasureRes.GroupByValues), len(queryMeasureRes.GroupByValues))
+		}
+
+		if verifyGroupByValues {
+			equal := reflect.DeepEqual(queryMeasureRes.GroupByValues, expMeasureRes.GroupByValues)
+			if !equal {
+				return fmt.Errorf("ValidateStatsQueryResults: GroupByCombination mismatch, expected: %+v, got: %+v", expMeasureRes.GroupByValues, queryMeasureRes.GroupByValues)
+			}
 		}
 
 		if len(queryMeasureRes.MeasureVal) != len(expMeasureRes.MeasureVal) {
 			return fmt.Errorf("ValidateStatsQueryResults: MeasureVal length mismatch, expected: %v, got: %v", len(expMeasureRes.MeasureVal), len(queryMeasureRes.MeasureVal))
 		}
 
-		for key, value := range expMeasureRes.MeasureVal {
-			actualValue, exist := queryMeasureRes.MeasureVal[key]
-			if !exist {
-				return fmt.Errorf("ValidateStatsQueryResults: MeasureVal not found for key: %v", key)
-			}
-			expFloatValue, isFloat := value.(float64)
-			if isFloat {
-				equal, err = CompareFloatValues(actualValue, expFloatValue)
-				if err != nil {
-					return fmt.Errorf("ValidateStatsQueryResults: Error comparing float values, err: %v", err)
+		if verifyGroupByValues {
+			for key, value := range expMeasureRes.MeasureVal {
+				actualValue, exist := queryMeasureRes.MeasureVal[key]
+				if !exist {
+					return fmt.Errorf("ValidateStatsQueryResults: MeasureVal not found for key: %v", key)
+				}
+				expFloatValue, isFloat := value.(float64)
+				if isFloat {
+					equal, err = CompareFloatValues(actualValue, expFloatValue)
+					if err != nil {
+						return fmt.Errorf("ValidateStatsQueryResults: Error comparing float values, err: %v", err)
+					}
+				} else {
+					equal = reflect.DeepEqual(actualValue, value)
+				}
+
+				if !equal {
+					return fmt.Errorf("ValidateStatsQueryResults: MeasureVal mismatch for key: %v, expected: %v, got: %v", key, value, actualValue)
 				}
-			} else {
-				equal = reflect.DeepEqual(actualValue, value)
 			}
+		} else {
+			// if group by values are not verified, then the bucket keys may not be consistent.
+			// So, we will only verify the measure values.
+			expMeasureSlice := utils.MapToSlice(expMeasureRes.MeasureVal)
```

Comment:
So basically, we are just checking the agg/measure names in `MeasureVal`, but the actual values are skipped because of inconsistency in timestamps due to different execution times?

Reply:
Yeah I changed this. We don't need to do this.